pil-analyzer/src/evaluator.rs

```diff
@@ -470,41 +471,54 @@ impl<'a> Definitions<'a> {
 impl<'a, T: FieldElement> SymbolLookup<'a, T> for Definitions<'a> {
     fn lookup(
         &mut self,
-        name: &str,
+        poly: &PolynomialReference,
         type_args: &Option<Vec<Type>>,
     ) -> Result<Arc<Value<'a, T>>, EvalError> {
-        Self::lookup_with_symbols(self.0, name, type_args, self)
+        let impl_pos = type_args
+            .as_ref()
+            .and_then(|type_args| poly.resolved_impls.get(type_args).as_ref().copied());
+
+        match impl_pos {
+            Some(index) => {
+                let parts = poly.name.split("::").collect::<Vec<_>>();
+                let fn_name = parts.last().unwrap().to_string();
+                let trait_name = parts[..parts.len() - 1].join("."); // TODO GZ Replace '.' by '::' after we merge #1694
+
+                let impls = match self.1.get(&trait_name) {
+                    Some(impls) => impls,
+                    None => Err(EvalError::SymbolNotFound(format!(
```

Comment:
I think you can use `self.1.get().unwrap()`, since the type checker (or some earlier component) guarantees that the trait is available, or does it not?